docs/design/node-selection.md

```diff
@@ -0,0 +1,82 @@
+
+The existing reputation-like system uses uptime and audit responses.  It does not currently consider geographic location, throughput, or latency.  In addition to factors which affect reputation, there are other factors in node selection.  These considerations currently include IP address, advertised available bandwidth, advertised available disk space, software version compatibility, and whether the node appeared to be online in the latest communication with the satellite.
+
+
+## Design
+
+Separate reputation scores will be maintained for audits and uptime.  Two sets of configurations will be stored, two different reputation scores will be calculated, etc..  
+
+The reputation _R(n)_ is calculated based on some measured success _v_, two persisted values a and β, and fixed configuration values λ and _w_.  
+
+> α(n) = λ·α(n-1) + _w_(1+_v_)/2
+>
+> β(n) = λ·β(n-1) + _w_(1-_v_)/2
+>
+> R(n) = α(n) / (α(n) + β(n))
+
+Initial values for a and β - a0 and β0 - will also be configuration values.  While _v_ may need to vary depending on our "easing" implementation, it will likely vary algorithmically and should not require configuration.  The initial implement may simply assume that _v_ = 1.
+
+Existing codes which updates database audit / uptime success-counts and ratios must be updated to track audit / uptime a and β values.  TotalAuditCount and TotalUptimeCount will still be needed to determine if the node is new or vetted.
+
+The node selection SQL queries will also change.  Twice as many nodes must returned from these functions to satisfy the "Power of Two Choices" requirement, which gives preference to nodes with better reputation.  For every two nodes returned, the one with the higher reputation scores will be selected returned and the other discarded.
+
+Note that the initial implementation has two different reputation statistics:  audit and uptime.  For the purposes of node selection, we assume that these two reputation can be combined by scaling one of them by some constant.  We further assume that operations may weigh these reputations differently.  The initial configuration should include a uptime_repair_weight, audit_repair_weight, uptime_uplink_weight, and audit_uplink_weight constants.
+
+> Total Repair Reputation = uptime_repair_weight · uptime R(n) + audit_repair_weight · audit R(n)
+>
+> Total Uplink Reputation = uptime_uplink_weight · uptime R(n) + audit_uplink_weight · audit R(n)
```

Comment:
The document needs some clarification about the 2 total reputations, the _repair_ and the _uplink_ one.

I may lack in how the audits and uptime checks are done, but I don't personally get this separation and specifically what the _uplink_ reputation is.

Reply:
Better?

Reply:
If you're referring to the variable formatting of the above paragraph, yes they are more readable, however it doesn't reply my doubt.

~~Meanwhile I understand the `Total Repair Reputation` as the reputation of the node when the satellite is doing repairs and use contact to that specific network (uptime) I'm wondering what an audit means in the context of a repair?~~

~~On the other hand, I don't know when a satellite does the uptime and audits for the `Total Uplink Reputation` calculation, because I don't understand what `Uplink` is in the Satellite context.~~

While writing the above, I've understood what this is, mostly there are 2 process, the uptime and audit which run in the satellite, then each time that both are completed, there will be registered this 2 different total reputations, one will be used for the repair process to select nodes based on the `Total Repair Reputation` and the other (`Total Uplink Reputation`) will be used for returning the list of nodes to uplink when it request orders to upload new files.